Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -95,16 +113,45 @@ class BTAnalyticsService: Equatable {
             if let http = self.http, http.baseURL.absoluteString == "test://do-not-send.url" {
                 return
             }
+            
+            if shouldBypassTimerQueue {
+                await self.sendQueuedAnalyticsEvents(configuration: configuration)
```

Comment:
what is this option? To turn off this implementation?

Reply:
Another question I had was right now in main, individual analytics event is sent to `v1/tracking/batch/events` which handles batching of the events on the server side?

Reply:
This just bypasses the timer to allow us to unit test sending analytics without needing to override the timer implementation or add a super long sleep.

Reply:
We should look into mocking DispatchTimerSource (either via subclassing or a protocol) and injecting an actual mock of it for our test setup, versus toggling this boolean. That would be the best way to test this, though it might not be possible with DispatchTimerSource. It is worth trying though - Jax did you take a stab at it?

Reply:
Yeah, I poked around at it a bit but mocking `DispatchTimerSource` from what I could find it was a bit odd because `DispatchTimerSource` is a protocol. It also was only mocking the timer methods (so schedule, setEventHandler, resume, cancel) vs also needing to send the event immediately. I think we'd have to have all the methods we need in our own protocol before going about the inheritance in `DispatchTimerSource`, then an extension for the implementation. We also would need to mock `DispatchQueue` which is used in the `makeTimerSource` method to send the event immediately.

But also maybe I over complicated it and missed something super obvious. 🙈 Happy to pair on it and see if we can get anything together if you want!